scripts/build.js

```diff
@@ -0,0 +1,110 @@
+const fs = require('fs');
+const path = require('path');
+
+const { ucs2 } = require("punycode"); 
+const SimpleIcons = require('simple-icons');
+const svg2ttf = require('svg2ttf');
+const SVGPath = require('svgpath');
+const ttf2woff = require('ttf2woff');
+const ttf2woff2 = require('ttf2woff2');
+
+
+const DISTDIR = path.resolve(__dirname, '..', 'font');
+
+const buildSimpleIconsSvgFontFile = () => {
+  let fileContent = '<?xml version="1.0" standalone="no"?>\n'
+                  + '<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\n'
+                  + '<svg xmlns="http://www.w3.org/2000/svg">\n'
+                  + '<metadata>CC0 1.0 Universal | simple-icons contributors</metadata>\n'
+                  + '<defs>\n'
+                  + '<font id="Simple Icons" horiz-adv-x="1200">\n'
+                  + '<font-face font-family="Simple Icons" units-per-em="1200" ascent="-1" descent="1200" />\n'
+                  + '<missing-glyph horiz-adv-x="0" />\n';
+  
+  let startUnicode = 0xea01;
+  let usedUnicodes = [];
+  
+  let unicodeHexBySlug = {}
+  
+  for (let iconTitle in SimpleIcons) {
+    let unicode = ucs2.decode(
+      String.fromCodePoint(startUnicode++)).map(
+        point => '&#x' + point.toString(16).toUpperCase() + ';')[0];
```

Comment:
This part is a bit hard to follow, and I would like to better understand what's going on. Are the suggestions below correct? If so, could you update this and the surrounding code?

```suggestion
    let nextUnicode = ucs2.decode(String.fromCodePoint(unicodeIndex++))[0];
    let unicodeString = nextUnicode.map(point => `&#x${point.toString(16).toUpperCase()};`);
```

Reply:
Yes, are correct, but I've improved it a bit:

```javascript
let nextUnicode = ucs2.decode(String.fromCodePoint(startUnicode++));
let unicodeString = nextUnicode.map(point => `&#x${point.toString(16).toUpperCase()};`).join('');
```
- I've used `startUnicode` instead of `unicodeIndex` because is an Unicode character in HEX format, not an index.
- This code is extracted from `svgicons2svgfont` (dependency of `icon-font-buildr`), from [here](https://github.com/nfroidure/svgicons2svgfont/blob/b8ed724ec20793edfeec913f7184fdbab16909f2/src/index.js#L488-L491), [here](https://github.com/nfroidure/svgicons2svgfont/blob/94968febca43b10e6b498a111198d70e1790c1ca/src/metadata.js#L61) and [here](https://github.com/nfroidure/svgicons2svgfont/blob/94968febca43b10e6b498a111198d70e1790c1ca/src/metadata.js#L15-L16). I'm not 100% sure it's correct, but the theory is to iterate through the unicode characters one by one and insert them encoded in the font file, one for each icon, then in the CSS insert them encoded with the format that CSS needs.

Reply:
> I've used startUnicode instead of unicodeIndex because is an Unicode character in HEX format, not an index.

I suggested changing the name since it used as a running "index"-like value, not just the starting value. Would still like to see that changed but indeed, index doesn't seem like the best word...


> This code is extracted from svgicons2svgfont (dependency of icon-font-buildr), from here, here and here. I'm not 100% sure it's correct, but the theory is to iterate through the unicode characters one by one and insert them encoded in the font file, one for each icon, then in the CSS insert them encoded with the format that CSS needs.

I figured something like that was going on. I think it is quite clear now what it does (i.e. iterating over unicode characters), I just think the original 3-line spanning statement was a bit much to parse.